stko/optimizers/macromodel.py

```diff
@@ -6,51 +6,29 @@
 
 """
 
-import os
-import subprocess as sp
-import time
+
 import rdkit.Chem.AllChem as rdkit
-import re
 from uuid import uuid4
 import logging
-import gzip
+
+
+from ..packages.macromodel import (
+    MacroModel,
+    MacroModelInputError,
+)
 from ..utilities import (
     MAEExtractor,
     move_generated_macromodel_files,
-    mol_from_mae_file
+    mol_from_mae_file,
 )
 from .optimizers import Optimizer
 
 logger = logging.getLogger(__name__)
 
 
-class MacroModelConversionError(Exception):
-    ...
-
-
-class MacroModelPathError(Exception):
-    ...
-
-
-class MacroModelForceFieldError(Exception):
-    ...
-
-
-class MacroModelOptimizationError(Exception):
-    ...
-
-
-class MacroModelLewisStructureError(Exception):
-    ...
-
-
-class MacroModelInputError(Exception):
-    ...
-
-
-class MacroModel(Optimizer):
+class MacroModelOptimizer(Optimizer, MacroModel):
```

Comment:
I like what you have done - but I think the `package` (I do not love that name, I prefer `wrapper` - but maybe we should discuss?) should be part of the init arguments of an Optimizer (see how an Extractor is contained in the Results class)